includes/class-wc-subscriptions-coupon.php

```diff
@@ -219,8 +219,24 @@ public static function get_discount_amount_for_cart_item( $cart_item, $discount,
 				}
 			}
 
-			// Apply sign-up discounts. Exclude switch cart items because their initial amount is entirely sign-up fees but should be treated as initial amounts
-			if ( ! $is_switch && WC_Subscriptions_Product::get_sign_up_fee( $cart_item['data'] ) > 0 ) {
+			// Compute the original sign-up fee. If it's a switch, we need to get the signup fee less
+			// upgrade costs.
+			if ( $is_switch ) {
+				$sign_up_fee_prorated = (int) $cart_item['data']->get_meta( '_subscription_sign_up_fee_prorated' );
+				$price_prorated       = (int) $cart_item['data']->get_meta( '_subscription_price_prorated' );
+
+				if ( 0 === $sign_up_fee_prorated && 0 === $price_prorated ) {
```

Comment:
Quick question for my understanding - why the `price_prorated` is also being checked here? 🤔

Reply:
Good question! In https://github.com/Automattic/woocommerce-subscriptions-core/pull/751/commits/3ee2abcea61ee0e03d133f591dc21d711323075a, I simplified the logic using `meta_exists`. Hopefully it's more readable.